core/src/main/java/org/infinispan/stream/impl/LockedStreamImpl.java

```diff
@@ -100,6 +104,15 @@ public void forEach(BiConsumer<Cache<K, V>, ? super CacheEntry<K, V>> biConsumer
    }
 
    @Override
+   public <R> Map<K, R> invokeAll(BiFunction<Cache<K, V>, ? super CacheEntry<K, V>, R> biFunction) {
+      Map<K, R> map = new HashMap<>();
+      Iterator<KeyValuePair<K, R>> iterator = realStream.map(new CacheEntryFunction<>(biFunction, predicate))
+            .filter(StreamMarshalling.nonNullPredicate()).iterator();
```

Comment:
`null` values are returned from the map. This is here for entries that existed before it was able to acquire the lock. https://github.com/infinispan/infinispan/pull/5456/files#diff-e7cee659aa5d59cab1f6af7972176bfaR213

`null` value is wrapped in the `KeyValuePair`

Reply:
The reason for `iterator` is because of the retry semantics between that and `collect`. The `iterator` method gurantees at least once for all elements (just like collect). However `iterator` will only cause `more than once` equal to `batchSize * numNodesLeft` due to it tracking by key. With `collect` you can have `more than once` equal to `allEntries * numTopologyUpdates`.